src/Export/Base/XlsxRenderer.php

```diff
@@ -39,6 +39,18 @@ protected function saveSpreadsheet(Spreadsheet $spreadsheet): string
             throw new \Exception('Could not open temporary file');
         }
 
+        // Enable auto filter
+        $sheet = $spreadsheet->getActiveSheet();
+        $sheet->setAutoFilter('A1:' . $sheet->getHighestColumn() . '1');
+
+        // Freeze first row
+        $sheet->freezePane('B2');
+
+        // Auto size columns to fit at least the headers
+        foreach (range('A', $sheet->getHighestColumn()) as $column) {
+            $sheet->getColumnDimension($column)->setAutoSize(true);
```

Comment:
Did you test how that looks, when a (ver)y long description is used?

Reply:
Ah good idea, nope not yet. Just squash it away, may be a better idea to lock it at the first row but I need to dig into that if excel supports that. As the implementing class is pretty generic it would be hard to inject proper "max width and wrap content" rules into it I think.

Reply:
I'd be very happy to get some help with that, as I have neither much experience with spreadsheets nor am I using it myself. 
I have implemented height/line-wrap for invoices [recently here](https://github.com/kevinpapst/kimai2/commit/72b73bc9a690435034e6613f44be0b7494eedbc0). Maybe something like this could be done for exports as well?

Reply:
`autosize` didn't work properly for me, maybe I did something wrong... or does it only works for the width and not for the height?

Reply:
Sure, I don't see why not. You use a specific renderer from the looks of it for invoices, right? I must dig a bit deeper into it. Just remove it from this one when you squash and I will take a look on thursday, I got a bit more time and brain there. Never used the invoices feature myself, will take a look into it.

PhpOffice is quite complete, I did not find any case where it did not sport a specific feature. On the other hand creating xlsx exports are a pain as you either have to throw around letter-coordinates or x/y based coordinates that are not fully feature complete in all functions.

Reply:
autosize is only for width sadly. If you want to handle height `$sheet->getDefaultRowDimension()->setRowHeight(-1);` might be a better solution.

Reply:
Gotta go now, here are some hints I collected for myself, maybe they help or give some clues what you could do as well. Got my through localization issues we had with german / english exports.

```
// Force text
$sheet->getStyle(PHPExcel_Cell::stringFromColumnIndex($index))
	->getNumberFormat()
	->setFormatCode(PHPExcel_Style_NumberFormat::FORMAT_TEXT);

// Proper number
$sheet->getStyle(PHPExcel_Cell::stringFromColumnIndex($index))
	->getNumberFormat()
	->setFormatCode('#,##0_-');

// Proper date
$sheet->getStyle(PHPExcel_Cell::stringFromColumnIndex($index))
	->getNumberFormat()
	->setFormatCode(PHPExcel_Style_NumberFormat::FORMAT_DATE_XLSX14);

// Fill with color
$style->getFill()->setFillType(PHPExcel_Style_Fill::FILL_SOLID)->setStartColor(
	new PHPExcel_Style_Color('00FFA200')
);

// Join multiple columns in first row of headers to group them for the second row that is split into more detailed rows (like logistic 1st row, pake width / height / depth in 2nd row).
$style->getFill()->setFillType(PHPExcel_Style_Fill::FILL_SOLID)->setStartColor(
	new PHPExcel_Style_Color('00C3C3C3')
);
$style->getFont()->setColor(new  PHPExcel_Style_Color(PHPExcel_Style_Color::COLOR_BLACK));
$sheet->mergeCellsByColumnAndRow($col, $row, $col + 4, $row);
```